test/new-e2e/examples/customenv_with_docker_app_test.go

```diff
@@ -0,0 +1,186 @@
+var lighttpdComposeContent string
+
+//go:embed testfixtures/lighttpd.conf
+var lighttpdConfigContent string
+
+//go:embed testfixtures/lighttpd_integration.conf.yaml
+var lighttpdIntegrationConfigContent string
+
+func vmPlusDockerEnvProvisioner() e2e.PulumiEnvRunFunc[vmPlusDockerEnv] {
+	return func(ctx *pulumi.Context, env *vmPlusDockerEnv) error {
+		awsEnv, err := aws.NewEnvironment(ctx)
+		if err != nil {
+			return err
+		}
+
+		// First we create a remote host with Amazon Linux ECS, that comes with Docker pre-installed
+		remoteHost, err := ec2.NewVM(awsEnv, "main", ec2.WithOS(os.AmazonLinuxECSDefault))
+		if err != nil {
+			return err
+		}
+		// we export it to env.RemoteHost, this will automatically initialize the ssh client on env.RemoteHost
+		remoteHost.Export(ctx, &env.RemoteHost.HostOutput)
+
+		// create a fakeintake instance on ECS Fargate
+		fakeIntake, err := fakeintake.NewECSFargateInstance(awsEnv, "")
+		if err != nil {
+			return err
+		}
+		err = fakeIntake.Export(ctx, &env.Fakeintake.FakeintakeOutput)
+		if err != nil {
+			return err
+		}
+
+		// Create a docker manager
+		dockerManager, _, err := docker.NewManager(*awsEnv.CommonEnvironment, remoteHost)
+		if err != nil {
+			return err
+		}
+		// let's create a lighttpd container
+		// first we create a config file and a directory for log files
+		// that we will mount in the container
+		// files will be created eventually, our docker compose needs to wait for them
+		// we track the creation of files in an array of pulumi resources
+		mountedFilesCommands := make([]pulumi.Resource, 0, 2)
+		// create a tmp directory on the remote host, we will use it to store the configuration file
+		// and to read log files
+		// lighttpdDirCmd is the command to create the directory
+		// it will be executed on the remote host, we can run docker compose
+		// after it is done
+		remoteTmpDirCmd, lighttpdDir, err := remoteHost.OS.FileManager().TempDirectory("lighttpd")
+		if err != nil {
+			return err
+		}
+		// export the remote directory path to the environment
+		env.remoteHostLogsDir = lighttpdDir
+		mountedFilesCommands = append(mountedFilesCommands, remoteTmpDirCmd)
+		// write the lighttpd configuration file
+		lighttpdConfigFile := lighttpdDir + "/lighttpd.conf"
+		lighttpdConfigFileCmd, err := remoteHost.OS.FileManager().CopyInlineFile(pulumi.String(lighttpdConfigContent), lighttpdConfigFile, false)
+		if err != nil {
+			return err
+		}
+		mountedFilesCommands = append(mountedFilesCommands, lighttpdConfigFileCmd)
+
+		// the name is internal only
+		envVars := pulumi.StringMap{
+			"DD_LIGHTTPD_CONFIG":    pulumi.String(lighttpdConfigFile),
+			"DD_LIGHTTPD_LOGS_PATH": pulumi.String(lighttpdDir),
+		}
+		// compose lighttpd
+		composeLighttpdCmd, err := dockerManager.ComposeStrUp("lighttpd", []docker.ComposeInlineManifest{
+			{
+				Name:    "lighttpd",
+				Content: pulumi.String(lighttpdComposeContent),
+			},
+		}, envVars, pulumi.DependsOn(mountedFilesCommands))
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities. -->
**💭 thought**
We could simplify this removing the name parameter and using a concat of compose manifest names